src/Extensions/JsConsoleLogger/JsConsoleLogger.php

```diff
@@ -70,19 +117,28 @@ private function checkJsErrors(StepEvent $event): void
      */
     private function writeLogFile(StepEvent $event, array $log): void
     {
-        file_put_contents($this->getLogFileName($event), print_r($log, true));
+        // logs need to be appended when test case uses data provider
+        file_put_contents($this->getLogFileName($event), print_r($log, true), FILE_APPEND);
     }
 
+    /**
+     * @param StepEvent $event
+     * @return string
+     */
     private function getLogFileName(StepEvent $event): string
     {
         return sprintf(
-            '%s/_output/%s.%s.fail.js-errors.txt',
-            dirname(__DIR__),
+            '%s/%s.%s.fail.js-errors.txt',
+            $this->getLogDir(),
             str_replace(['\\', ':'], '.', $this->getTestName($event)),
             $this->environment
         );
     }
 
+    /**
+     * @param StepEvent $event
+     * @return string
+     */
```

Comment:
And again same here please. Can you remove the superfluous @param and @return in the docblock they are already specified in the code. Thanks